spec/models/shared_examples/shared_examples_for_loadable.rb

```diff
@@ -43,6 +43,28 @@
         expect(results.num_inserts).to eq(1)
         expect(results.ids.length).to eq(1)
       end
+
+      it 'does roll back to the old table content if the upload is invalid' do
+        allow(described_class).to receive(:load_csv_file).and_raise('error in csv')
+        before_count = described_class.count
+        begin
+          described_class.load(csv_file_invalid, load_options)
```

Comment:
you can avoid the begin/rescue with `expect{described_class.load(csv_file_invalid, load_options)}.to raise_error`